packages/in_app_purchase/ios/Classes/FIAPaymentQueueHandler.m

```diff
@@ -57,7 +58,9 @@ - (void)finishTransaction:(SKPaymentTransaction *)transaction {
 - (void)paymentQueue:(SKPaymentQueue *)queue
     updatedTransactions:(NSArray<SKPaymentTransaction *> *)transactions {
   for (SKPaymentTransaction *transaction in transactions) {
-    [self.transactionsSetter setObject:transaction forKey:transaction.transactionIdentifier];
+    if (transaction.transactionIdentifier) {
```

Comment:
Would this ever be null? Should we handle it like an error if/when it is?

Reply:
Apple handles the transaction identifier strangely, the identifier is null when the state the purchasing. I should probably put a comment on it.